src/julia_internal.h

```diff
@@ -1117,4 +1117,35 @@ jl_assume_aligned(T ptr, unsigned align)
 }
 #endif
 
+// -- typemap.c -- //
+
+typedef struct _jl_frontend_t {
```

Comment:
Yes, the API surface is too large. I just wanted to get everything separated out for now. This definitely needs to get refactored.

Reply:
Agreed, while it's good to see where this might be going, it would be nice to break up this PR into many smaller and independent pieces so that the non functional changes (like splitting up files) happens in separate commits.

Reply:
The splitting up is already in a separate commit.

Reply:
I assume you mean 5f3216859f2134ffd6572c699f14aa44bf070433, but that's not NFC.

Reply:
Depends on your definition of NFC. It does add the jl_frontend interface, but it's mostly code movement.

Reply:
Adding an internal abstract interface + code movement seems like NFC to me.

Reply:
It would be good to get 5f32168 in (maybe as a couple of commits). Cherry picking that now.

Reply:
See #32063